packages/node/src/utils.ts

```diff
@@ -1,8 +1,37 @@
 import { debug, streamToBuffer } from '@vercel/build-utils';
-import { pathToRegexp } from 'path-to-regexp';
+import { pathToRegexp as pathToRegexpCurrent, Key } from 'path-to-regexp';
+import { pathToRegexp as pathToRegexpUpdated } from 'path-to-regexp-updated';
 import type { IncomingMessage } from 'http';
 import { extname } from 'path';
 
+// run the updated version of path-to-regexp, compare the results, and log if different
+function pathToRegexp(
+  callerId: string,
+  path: string,
+  keys?: Key[],
+  options?: { strict: boolean; sensitive: boolean; delimiter: string }
+) {
+  const currentRegExp = pathToRegexpCurrent(path, keys, options);
+
+  try {
+    const newRegExp = pathToRegexpUpdated(path, keys, options);
+    const diff = currentRegExp.toString() !== newRegExp.toString();
+    // FORCE_PATH_TO_REGEXP_LOG can be used to force these logs to render
+    // for verification that they show up in the build logs as expected
+    if (process.env.FORCE_PATH_TO_REGEXP_LOG || diff) {
+      // each of these path-to-regexp comparisons logs an innocuous message
+      // so we (1) can easily grab it by querying the build logs
+      // and (2) it doesn't spook the user
+      console.log(`SYSTEM TEST #${callerId} A: ${path}`);
```

Comment:
Prefix the log with `[vc]` and it'll be omitted from the user-facing logs entirely (but still be queryable on the backend).

Reply:
Also, should we log the before and after regexps?